Documentation/design-docs/standalone-gc-eventing.md

```diff
@@ -0,0 +1,212 @@
+# Standalone GC Eventing Design
+
+Author: Sean Gillespie (@swgillespie) - 2017
+
+This document aims to provide a specification for how a standalone GC fires
+events that can be collected by trace collectors. Such a feature is highly desirable
+for a standalone GC since it is the primary way that is used to reason about
+GC performance. Since a standalone GC is not permitted to link against the rest of the runtime, all
+communication betwene the runtime and the GC most pass through dynamically-dispatched interfaces.
+
+## Definitions
+
+* An **event** is some unit of information that the runtime can issue if requested. In general,
+this is used for lightweight tracing. Managed code can issue events using
+`System.Diagnostics.Tracing.EventSource`. Native code (i.e. the runtime) issues events by calling macros
+that delegate the issuing of events to autogenerated code that is generated to interface with the
+underlying event implementation. Events are only issued if they are turned on; the mechanism by which
+events are turned on is not in the scope of this document
+* The **payload** of an event is some amount of data that is delivered with the event itself. Its size
+may be variable. Most events that are fired by the runtime have a schema (predefined layout), but it
+is not a requirement.
```

Comment:
The schema is technically required to be known at build time for LTTng.  All of the other loggers don't necessarily require that the schema be known at build time, but they do require it to be known at analysis time.